### PR TITLE
Log customization doc: ParseStateValues

### PR DESCRIPTION
**Pull request by TimothyMothra**
#2999

## Changes
- section for `OpenTelemetryLoggerOptions.ParseStateValues`
- code examples copied from extending-the-sdk

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

File: docs/logs/customizing-the-sdk/README.md [hunk 1]
```diff
@@ -32,7 +32,14 @@ set by invoking the formatter from [ILogger.Log](https://docs.microsoft.com/dotn
 
 ### ParseStateValues
 
-TODO
+`ILogger` supports structured logging which can assign either types or
+[message templates](https://docs.microsoft.com/dotnet/core/extensions/logging#log-message-template)
+to the log state.
+
+`ParseStateValues` indicates if the log state will be parsed into
```
Comment (reyang) [inline docs/logs/customizing-the-sdk/README.md:39]: What does this actually mean?

File: docs/logs/customizing-the-sdk/Program.cs [hunk 2]
```diff
@@ -39,15 +42,39 @@ public static void Main()
...
 
+        // unstructured log
         logger.LogInformation("Hello Information");
         logger.LogWarning("Hello Warning");
         logger.LogError("Hello Error");
 
+        // structured log with template
+        logger.LogInformation("Hello from {name} {price}.", "tomato", 2.99);
+
+        // structured log with strong type
```
Comment (cijothomas) [inline docs/logs/customizing-the-sdk/Program.cs:53]: How is this related to the feature being documented?

Comment (TimothyMothra) [reply docs/logs/customizing-the-sdk/Program.cs:53]: I included it here for completeness, because the type is part of the state object. 
Although it's not handled well by the ConsoleExporter.

I can omit this and focus on the simpler example using the message template

File: docs/logs/customizing-the-sdk/README.md [hunk 3]
```diff
@@ -32,7 +32,13 @@ set by invoking the formatter from [ILogger.Log](https://docs.microsoft.com/dotn
 
 ### ParseStateValues
 
-TODO
+ILogger supports [message templates](https://docs.microsoft.com/dotnet/core/extensions/logging#log-message-template)
```
Comment (cijothomas) [inline docs/logs/customizing-the-sdk/README.md:35]: this is irrelevant.
check the code doc for `ParseStateValues`, and copy what we have there.

Comment (TimothyMothra) [reply docs/logs/customizing-the-sdk/README.md:35]: The code doc states: "Gets or sets a value indicating whether or not log state should be parsed into LogRecord.StateValues on generated LogRecords."

I don't think this does a good job of explaining what the log state is or what data will be set into LogRecord.StateValues.

Here, I was trying to introduce what the log state is and what parts of it are set in the StateValues.
